scripts/libraries/rtsp.py

```diff
@@ -274,28 +323,45 @@ def __send_rtp(self, image_callback, quality):  # private
                     jpeg_header = struct.pack(
                         ">IBBBB", i, 0, quality, int(img.width() // 8), int(img.height() // 8)
                     )
-                    img_data = img.bytearray()[i : i + min(l, max_packet_size)]
+                    img_data = mv[i : i + min(l, max_packet_size)]
                     img_data_len = len(img_data)
-                    data_len = self.__send(rtp_header + jpeg_header + img_data)
-                    if not data_len:
+                    if not self.__send(rtp_header + jpeg_header + img_data):
```

Comment:
There's a chance here to optimize GC/RAM usage, if you split this into 3 send calls no new memory will be allocated, currently I think it makes a new object with headers + image data. Also for the above struct, if you use a bytearray instead and just change the `i` bytes, no new memory will be created every iteration. It's fine for now but might want to fix it in the future.